akka-javasdk-tests/src/test/java/akkajavasdk/components/agent/AgentIntegrationTest.java

```diff
@@ -130,6 +130,58 @@ public void shouldCallToolFunctions() {
     assertThat(response.response()).isEqualTo("The weather is sunny in Leuven. (date=2025-01-01)");
   }
 
+  @Test
+  public void shouldCallToolFunctionsFromClasses() {
+
+    var userQuestion = "How is the traffic today in Leuven?";
+
+    var args = """
+          {
+           "location" : "Leuven"
+          }
+          """;
+    // when asking for the traffic, call the traffic service
+
+    testModelProvider.mockToolInvocationRequest(
+      msg -> msg.content().equals(userQuestion),
+        new TestModelProvider.ToolInvocationRequest("getTrafficNow", args));
+
+    // receives the traffic info as final answer
+    testModelProvider.mockResponseToToolResult(
+      result -> result.content().startsWith("There is traffic jam"),
+      result -> new TestModelProvider.AiResponse(result.content())
+    );
+
+    //when
+    var response = componentClient.forAgent().inSession(newSessionId())
+      .method(SomeAgentWithTool::query)
+      .invoke(userQuestion);
+
+
+    //then
+    assertThat(response.response()).isEqualTo("There is traffic jam in Leuven.");
+  }
+
+  @Test
+  public void shouldFailWithClearMessageIfToolClassCannotBeInit() {
+
+    var userQuestion = "How is the traffic today in Leuven?";
+
+    testModelProvider.mockToolInvocationRequest(
+      msg -> msg.content().equals(userQuestion),
+      new TestModelProvider.ToolInvocationRequest("getNonStaticTrafficNow", ""));
+
+    try {
+      componentClient.forAgent().inSession(newSessionId())
+        .method(SomeAgentWithTool::query)
+        .invoke(userQuestion);
+
+      fail("Should have thrown an exception");
+    } catch (Exception e) {
+      // FIXME: errors message in dev-mode/test should be propagate
+      assertThat(e.getMessage()).contains("Unexpected error");
```

Comment:
We are not getting the full error message here. I believe the runtime is only propagating for context in a few cases (ESE and Workflow), but not for Agents. 

We can improve it in follow-up.